prov/rxm/src/rxm_init.c

```diff
@@ -49,16 +60,19 @@ int rxm_info_to_core(uint32_t version, struct fi_info *hints,
 			core_info->domain_attr->mr_mode =
 				hints->domain_attr->mr_mode;
 
-			if (FI_VERSION_GE(version, FI_VERSION(1, 5)))
-				core_info->domain_attr->mr_mode |=
-					(FI_MR_LOCAL | OFI_MR_BASIC_MAP);
-
 			if (hints->domain_attr->caps)
 				core_info->domain_attr->caps =
 					hints->domain_attr->caps | FI_MSG;
 			if (hints->domain_attr->mode)
 				core_info->domain_attr->mode = hints->domain_attr->mode;
 		}
+	} else {
+		/* Since hints is NULL fake support for FI_MR_BASIC to allow
+		 * discovery of core providers like verbs which require it */
+		if (FI_VERSION_GE(version, FI_VERSION(1, 5)))
+			core_info->domain_attr->mr_mode |= OFI_MR_BASIC_MAP;
+		else
+			core_info->domain_attr->mr_mode = FI_MR_BASIC;
```

Comment:
I think this can be UNSPEC, which is allowed to return either basic or scalable. Otherwise, you won't pick up scalable providers or use that feature if the provider does support it.